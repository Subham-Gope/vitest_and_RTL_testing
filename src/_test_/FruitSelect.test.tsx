// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import FruitSelect from "../components/FruitSelect";

// describe("FruitSelect.tsx", () => {
//   test("let's you select Banana", async () => {
//     const { container } = render(<FruitSelect />);

//     expect(container).toMatchSnapshot();

//     const select = screen.getByRole("combobox", { name: /fruits option/i });
//     expect(select).toBeInTheDocument();

//     await userEvent.selectOptions(select, "Banana");

//     const message = screen.getByText("Selected: Banana");
//     expect(message).toBeInTheDocument();
//   });
// });
