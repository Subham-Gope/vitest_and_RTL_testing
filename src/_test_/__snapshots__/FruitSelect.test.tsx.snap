// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FruitSelect.tsx > let's you select Banana 1`] = `
<div>
  <div
    class="h-screen w-screen flex"
  >
    <div
      class="flex flex-col gap-4 items-center justify-center p-12 m-auto bg-amber-5 w-[300px] h-[300px] bg-red-300"
    >
      <label
        class="text-3xl text-gray-50"
        for="fruits"
      >
        Fruits Option
      </label>
      <select
        class="text-xl text-gray-100 bg-gray-800 p-4 rounded"
        id="fruits"
        name="Fruits"
        required=""
      >
        <option
          class="text-[16px] text-gray-400"
          disabled=""
          value=""
        >
          Select a Fruit
        </option>
        <option
          class="text-[16px] "
          value="Banana"
        >
          Banana
        </option>
        <option
          class="text-[16px]"
          value="Apple"
        >
          Apple
        </option>
        <option
          class="text-[16px]"
          value="Jack Fruit"
        >
          Jack Fruit
        </option>
        <option
          class="text-[16px]"
          value="Dragon Fruit"
        >
          Dragon Fruit
        </option>
      </select>
    </div>
  </div>
</div>
`;
